src/node_messaging.cc

```diff
@@ -544,36 +544,87 @@ std::unique_ptr<MessagePortData> MessagePort::Detach() {
 }
 
 
-void MessagePort::Send(Message&& message) {
-  Mutex::ScopedLock lock(*data_->sibling_mutex_);
-  if (data_->sibling_ == nullptr)
-    return;
-  data_->sibling_->AddToIncomingQueue(std::move(message));
-}
+Maybe<bool> MessagePort::PostMessage(Environment* env,
+                                     Local<Value> message_v,
+                                     Local<Value> transfer_v) {
+  auto isolate = env->isolate();
+  auto obj = object(isolate);
+  auto context = obj->CreationContext();
```

Comment:
Can you avoid `auto` for all of these?

Reply:
But… `auto` is too exciting!

For reals though, I can change it of course, but I personally thought the variable names were pretty self explanatory in terms of types.

Reply:
They are, except maybe in the case of `obj` … but I’d personally really prefer to keep `auto` for cases where the exact type is hard to spell out/unknown – obviously, other people will have different experiences, but for me using `auto` usually makes writing the code easier and reading it harder, even in these “trvial” cases…

Reply:
I think mostly this code is read a lot more than it is written - while I use `auto` in my own code in Node.js I had zero (well ~2) C++ commits but I've reviewed at least a hundred PRs. So 👍 on spelling up the types.